stan/math/rev/fun/cholesky_decompose.hpp

```diff
@@ -331,34 +336,30 @@ class cholesky_opencl : public vari {
       const int k_j_ind = k - j;
       const int m_k_ind = M_ - k;
 
-      matrix_cl<var> R(k_j_ind, j, matrix_cl_view::Lower);
-      matrix_cl<var> D(k_j_ind, k_j_ind, matrix_cl_view::Lower);
-      matrix_cl<var> B(m_k_ind, j);
-      matrix_cl<var> C(m_k_ind, k_j_ind, matrix_cl_view::Lower);
-
-      R.sub_block(L, j, 0, 0, 0, k_j_ind, j);
-      D.sub_block(L, j, j, 0, 0, k_j_ind, k_j_ind);
-      B.sub_block(L, k, 0, 0, 0, m_k_ind, j);
-      C.sub_block(L, k, j, 0, 0, m_k_ind, k_j_ind);
+      auto&& R_val = block(L_val, j, 0, k_j_ind, j);
+      auto&& R_adj = block(L_adj, j, 0, k_j_ind, j);
+      matrix_cl<double> D_val = block(L_val, j, j, k_j_ind, k_j_ind);
+      matrix_cl<double> D_adj = block(L_adj, j, j, k_j_ind, k_j_ind);
+      auto&& B_val = block(L_val, k, 0, m_k_ind, j);
+      auto&& B_adj = block(L_adj, k, 0, m_k_ind, j);
+      auto&& C_val = block(L_val, k, j, m_k_ind, k_j_ind);
+      auto&& C_adj = block(L_adj, k, j, m_k_ind, k_j_ind);
 
-      C.adj() = C.adj() * tri_inverse(D.val());
-      B.adj() = B.adj() - C.adj() * R.val();
-      D.adj() = D.adj() - transpose(C.adj()) * C.val();
+      C_adj = C_adj * tri_inverse(D_val);
+      B_adj = B_adj - C_adj * R_val;
+      D_adj = D_adj - transpose(C_adj) * C_val;
 
-      symbolic_rev(D);
+      symbolic_rev(D_val, D_adj);
 
-      R.adj() = R.adj() - transpose(C.adj()) * B.val() - D.adj() * R.val();
-      D.adj() = diagonal_multiply(D.adj(), 0.5);
+      R_adj = R_adj - transpose(C_adj) * B_val - D_adj * R_val;
+      D_adj = diagonal_multiply(D_adj, 0.5);
 
-      L.adj().sub_block(R.adj(), 0, 0, j, 0, k_j_ind, j);
-      L.adj().sub_block(D.adj(), 0, 0, j, j, k_j_ind, k_j_ind);
-      L.adj().sub_block(B.adj(), 0, 0, k, 0, m_k_ind, j);
-      L.adj().sub_block(C.adj(), 0, 0, k, j, m_k_ind, k_j_ind);
+      block(L_adj, j, j, k_j_ind, k_j_ind) = D_adj;
```

Comment:
The rest are now handled by the kernel generator right?

Reply:
The rest use  only kernel gneerator functionis, so tzhey can use the blocks in place. This one uses some other functions that only work on matrix_cl, so we still need copys.